### ISSUE TITLE
Perks - Trainable / Teachable

### ISSUE THREAD

**Issue by Nelaif** [luttje/gmod-experiment-redux#70]
I thought about perks in the past alot. 
Maybe upon character creation you can choose 5 perks which you master. And you can teach people this 'skill' (carrying more stuff or being better at scrapping and stuff).

Like you can give 1 skill point every hour to somebody. You need like 30 points to achieve mastery in this skill. 
This will make the game more social. People might sell their skill points for bolts. 

Of course i like the idea of buying auguments for your character, but teachable skills could be next level. And your character is not able to learn all skills, so there would be different builds and possibilities per character and you will need to team up with people to achieve greater things. 

Like creating an exoskeleton for like 30% cheaper? You need the knowledge, you cant buy it.

**Comment by luttje** [comment]
Heya, thanks for contributing. I really like the idea! 

I think the perk system could use a big upgrade, because the current "big list of perks you can buy and they magically improve stuff for you" is a bit odd.

The way your idea encourages players to become more social is a great one too. We really do need that!

For now I'll mark this as something for the future 'open beta' test. We can use this issue until then to brainstorm further ideas.

I have some notes (mostly for myself) that I can't really form into a coherent reply atm, so I'll list them as is:
- [ ] Perks appear out of nowhere, kinda silly, let's find an in-world reason how they can work. Trainable skills can actually make sense over "magic game perk gives bonus"
- [ ] The big perk list is hard to navigate and overwhelming, I'm thinking a skill tree of sorts is more intuitive
- [ ] All perks can currently be purchased without restriction, should we limit it?
- [ ] Perks can be purchased with bolts, I like the skill point system mentioned more

**Comment by Nelaif** [comment]
Super nice , by the way are you Dutch by any chance?

Op di 30 apr 2024 om 15:39 schreef luttje ***@***.***>:

**Comment by luttje** [comment]
The reason I got rid of the karma system was that it felt obtuse. If you weren't the right karma you'd just buy tomes to change it. Even without the tomes it lured people into killing other karma characters, just to adjust their own karma to their liking.

Character hopping shouldn't be an issue, I've moved it to 1 character per player. So you'd need multiple steam accounts if you do want to manage that.

As for the skill tree being too much, it probably is. The reason I mentioned is, is that a "tree" can clearly display which perks are available, and how they're limited by progression for example. Another (simpler) approach could be to let players choose perks into a screen like this:
![](https://github.com/luttje/gmod-experiment-redux/assets/2738114/77335c7d-4c2c-4e11-b63c-06b594efd1c0)

This is all to hopefully demonstrate clearly to the player: you have 5 perks that can have a positive effect on others. You can have 10 other skills that help you. (The amounts can change)

**Comment by Nelaif** [comment]
Yes I think something progressive could be a nice addition. Like crafting medkits for example. Maybe you need some raw materials if you got 10/30 points. If you have 20/30 you can make them straight from bolts. 30/30 could be the ability to create a health station or survive a traumatic limb fracture or whatever.

It seems you're creating things quickly. I should visit the server to see how it all is and then I could make a list of ideas

**Comment by luttje** [comment]
Still love the idea, but I'll lower the priority slightly for now to keep a realistic goal

**Comment by Nelaif** [comment]
You are doing great things already!
